src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -496,12 +493,12 @@ public function editAction(Request $request, int $productId): Response
     public function deleteAction(int $productId): Response
     {
         try {
-            $this->getCommandBus()->handle(new DeleteProductCommand($productId));
+            $this->getCommandBus()->handle(new DeleteProductCommand($productId, ShopConstraint::allShops()));
             $this->addFlash(
                 'success',
                 $this->trans('Successful deletion', 'Admin.Notifications.Success')
             );
-        } catch (ProductException $e) {
+        } catch (Exception $e) {
```

Comment:
```suggestion
        } catch (ProductException $e) {
```
Catching `ProductException` here is on purpose, this represents the scope of exception that we are expecting and can handle an appropriate error message for. Anything out of this scope is probably unexpected and should fail with lots of noise 😅

That's actually why I suggested using our core `InvalidArgumentException` instead of the generic one.

Reply:
I remember we used different approach.

If we are not catching all exceptions, then we leave the possibility for errors to bubble up and break the page.
On the other hand, when we catch all of them, we have a fallback which at least shows any unmapped error in flash message "unexpected error bla bla", but doesn't break the page (which for me seems to be what we want, no?).
I remember some time ago this was required and even a dedicated PR was done to replace all the cought exceptions to generic ones and main argument was that even if we know what exceptions our core throws, we cannot know about module exceptions.

Btw in this specific case it can easily throw ShopException like (ShopAssociationNotFoundException, which is not ProductException, but a sub of ShopException)

Reply:
We could always `catch (ProductException | ShopException $e)` it works very well

But your argument about exceptions make sense, although it bothers me that we catch such wide Exception I remember this conversation about favoring stability and catching error with a flash message